src/org/etools/j1939tools/modules/CSERSModule.java

```diff
@@ -0,0 +1,133 @@
+                { "| Time Until FTP Engine Output Energy,                                    s  |", "SPN_22235", "SPN_22245" },
+                { "| Time Until Catalyst Cold Start Tracking Temperature Threshold,          s  |", "SPN_22236", "SPN_22246" },
+                { "|----------------------------------------------------------------------------+", "-------------+", "-------------+" },
+        };
+
+        return printTable(packets, headerRows, columnWidths, leftPad, table);
+    }
+
+    private String printTable(List<GenericPacket> packets,
+                              int headerRows,
+                              int[] columnWidths,
+                              boolean[] leftPad,
+                              String[][] table) {
+        StringBuilder sb = new StringBuilder();
+        for (int rowIndex = 0; rowIndex < table.length; rowIndex++) {
+            String[] row = table[rowIndex];
+            for (int colIndex = 0; colIndex < row.length; colIndex++) {
+                String cell = row[colIndex];
+                int columnWidth = columnWidths[colIndex];
+                if (cell.contains("SPN_")) {
+                    int spnId = parseSpnId(cell);
+                    String value = getSpnValue(packets, spnId);
+                    sb.append(StringUtils.padLeft(value, columnWidth)).append(" |");
+                } else {
+                    if (rowIndex < headerRows) {
+                        sb.append(StringUtils.center(cell, columnWidth));
+                    } else {
+                        if (leftPad[colIndex]) {
+                            sb.append(StringUtils.padRight(cell, columnWidth));
+                        } else {
+                            sb.append(StringUtils.padLeft(cell, columnWidth));
+                        }
+                    }
+                }
+            }
+            sb.append(NL);
+        }
+
+        String s = sb.toString();
+        return s;
+    }
+
+    private int parseSpnId(String cell) {
+        return Integer.parseInt(cell.replace("SPN_", "").replace(" ", ""));
+    }
+
+    private String getSpnValue(List<GenericPacket> packets, int spnId) {
+        return packets.stream()
+                .map(p -> p.getSpn(spnId))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(this::printSpn)
+                .map(v -> {
+                    if ("Not Available".equals(v)) {
+                        return "N/A";
+                    } else {
+                        return v;
+                    }
+                })
+                .findFirst()
+                .orElse("");
+    }
+
+    private String printSpn(Spn spn) {
+        if (spn.isNotAvailable() || spn.isError()) {
+            return spn.getStringValue();
+        }
+
+        double value = spn.getValue();
+
+        String unit = spn.getSlot().getUnit();
```

Comment:
Verify this.

Reply:
Latest commit updates unit in table from 's' to 'min' to match calculated value. Minute chosen for better readability.